src/app/Plugins/Weather/Components/weather/weather.component.ts

```diff
@@ -26,6 +29,20 @@ export class WeatherComponent implements OnInit {
     this.today = false;
     this.tommorow = false;
     this.week = false;
+    this.weatherService.getAllLocations()
+      .subscribe((data: Location[]) => {
+        this.locations = data;
+        this.selectedLocation = this.locations[0].name;
```

Comment:
does this throw an exception if the array is empty or is this fine?

Reply:
this can throw `TypeError: Cannot read property 'name' of undefined` if the index does not exists, then it returns `undefined` and `undefined`  dont have a `name`

Reply:
even if the array is empty, ngFor will just not iterate and the content will be empty